libs/platform/ebpf_object.c

```diff
@@ -317,8 +348,9 @@ ebpf_object_reference_next_object(
     ebpf_id_t start_id = (previous_object) ? previous_object->id : 0;
     ebpf_core_object_t* object = _get_next_object_by_id(start_id, type);
```

Comment:
nit: should we move the step of calling `_ebpf_object_try_acquire_reference()` inside `_get_next_object_by_id()`? This will ensure any future code change does not use the regular acquire_reference function when it calls `_get_next_object_by_id()`?